libmachine/provision/utils.go

```diff
@@ -229,3 +238,17 @@ func waitForDocker(p Provisioner, dockerPort int) error {
 
 	return nil
 }
+
+//fakeSSHCommander is intended to be used by testing code
+//Extend it when needed
+type fakeSSHCommander struct {
```

Comment:
Can you please make this exported (i.e. named `FakeSSHCommander` instead of `fakeSSHCommander`) and move it's definition into a test with `_test.go` name to delineate it from non-test code?  One suggestion is that you could add a `provisiontest` module inside of `provision` module (like how the standard library has `httptest`) and put it in a file there.